### ISSUE TITLE
Loosen Web3 from 4.0b7 to >=4.0.0,<5.0.0

### ISSUE THREAD

**Issue by stefanmendoza** [carver/ethtoken.py#2]
While working on https://github.com/ethereum/web3.py/issues/1056 and realizing eth-abi needs to have a patch (1.2.1) to include https://github.com/ethereum/eth-abi/pull/100, I also noticed a constraint being violated due to the latest version of ethtoken being pulled in here:

https://github.com/ethereum/web3.py/blob/v4.7.2/setup.py#L26

since the latest version of ethtoken has web3 locked - `web3~=4.0b7`.

I created a local patch of eth-abi-1.2.1:

```
[10:53:34 PM] @stefanmendoza ➜  web3.py git:(issue-1086) ✗ git diff
diff --git a/setup.py b/setup.py
index 8c4a8a4..605e0c8 100644
--- a/setup.py
+++ b/setup.py
@@ -24,6 +24,7 @@ extras_require = {
         #"eth-testrpc>=0.8.0",
         #"ethereum-tester-client>=1.1.0",
         "ethtoken",
+        # "ethtoken>=0.0.1-alpha.4,<2.0.0",
         "py-geth>=1.4.0",
         "py-solc>=0.4.0",
         "pytest>=2.7.2",
@@ -70,7 +71,7 @@ setup(
     install_requires=[
         "toolz>=0.9.0,<1.0.0;implementation_name=='pypy'",
         "cytoolz>=0.9.0,<1.0.0;implementation_name=='cpython'",
-        "eth-abi>=1.2.0,<2.0.0",
+        "eth-abi>=1.2.1,<2.0.0",
         "eth-account>=0.2.1,<0.4.0",
         "eth-utils>=1.0.1,<2.0.0",
         "hexbytes>=0.1.0,<1.0.0",
```
```
[10:53:35 PM] @stefanmendoza ➜  web3.py git:(issue-1086) ✗ pip list
Package      Version Location                                          
------------ ------- --------------------------------------------------
cytoolz      0.9.0.1 
eth-abi      1.2.1   /Users/stefanmendoza/Github/Forks/ethereum/eth-abi
eth-hash     0.2.0   
eth-typing   1.3.0   
eth-utils    1.2.2   
parsimonious 0.8.0   
pip          18.1    
setuptools   40.4.3  
six          1.11.0  
toolz        0.9.0   
wheel        0.32.1  
```

and only got this compatability error:
```
ethtoken 0.0.1a3 has requirement web3~=4.0b7, but you'll have web3 4.7.1 which is incompatible.
```

so I bumped this up

```
--- a/setup.py
+++ b/setup.py
@@ -38,7 +38,7 @@ extras_require['dev'] = (
 )
 
 install_requires = [
-    'web3~=4.0b7',
+    'web3>=4.0.0,<5.0.0',
 ]
 if sys.platform == 'win32':
     install_requires.append('pypiwin32')
@@ -46,7 +46,7 @@ if sys.platform == 'win32':
 setup(
     name='ethtoken',
     # NOTE: do not change version manually, use `make release bump=$$PART_TO_BUMP$$` instead
-    version='0.0.1-alpha.3',
+    version='0.0.1-alpha.4',
     description="""Ethereum EIP20 Token Interface""",
     long_description_markdown_filename='README.md',
     author='Jason Carver',
```

which got rid of the compatibility error. Not sure if we want to open it up to all 4.x, but since we're on 4.7.2 I think we can probably move off the beta now? I'm thinking once we release eth-abi-1.2.1 and potentially open up the version constraints on web3 here that I'll be unblocked on https://github.com/ethereum/web3.py/issues/1056. Thoughts?

**Comment by stefanmendoza** [comment]
It is interesting that we have a circular dependency on Web3 here. I think the order needs to be:
```
1. eth-abi-1.2.1 is released and includes the exclusion of eth-typing >= 2
2. web3-v4.7.2 will pick this up as it's >=1.2.0
3. Bump the version here to include non-beta 4.x Web3, which should include the change above. Release.
4. Web3 will pull in the latest version of ethtoken-0.0.1-alpha.4, which will not have an incompatibility with the Web3 version being installed.
```

**Comment by carver** [comment]
Thanks! Merged and released as ethtoken `v0.0.1-alpha.4`
